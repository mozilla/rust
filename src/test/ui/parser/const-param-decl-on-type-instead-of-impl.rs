struct NInts<const N: usize>([u8; N]);
impl NInts<const N: usize> {} //~ ERROR unexpected `const` parameter declaration

fn main() {
    let _: () = 42; //~ ERROR mismatched types
}
