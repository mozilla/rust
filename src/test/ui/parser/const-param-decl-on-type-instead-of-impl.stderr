error: unexpected `const` parameter declaration
  --> $DIR/const-param-decl-on-type-instead-of-impl.rs:2:18
   |
LL | impl NInts<const N: usize> {}
   |                  ^ expected a `const` expression, not a parameter declaration
   |
help: `const` parameters must be declared for the `impl`
   |
LL | impl<const N: usize> NInts<N> {}
   |     ^^^^^^^^^^^^^^^^       ^

error[E0308]: mismatched types
  --> $DIR/const-param-decl-on-type-instead-of-impl.rs:5:17
   |
LL |     let _: () = 42;
   |            --   ^^ expected `()`, found integer
   |            |
   |            expected due to this

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
